pandas/tests/indexes/test_base.py

```diff
@@ -1557,6 +1557,41 @@ def test_string_index_repr(self):
 
                 self.assertEqual(coerce(idx), expected)
 
+    def test_map(self):
+
+        # Applying a function to the Index
+        df = pd.DataFrame([[0, 1], [2, 3]], columns=['c1', 'c2'], index=['i1', 'i2'])
+        df.index.name = "Numbering"
+        df.index = df.index.map(lambda x: x.upper())
```

Comment:
Ok finally able to get back to this. Will a very simple test be enough than (like -> map with identity function returns an Index)? Otherwise it'll need a lot of `if isinstance`...